ModLoader/ModFilesystem.cpp

```diff
@@ -0,0 +1,194 @@
+#include "ModFilesystem.h"
+
+#include <algorithm>
+#include <ranges>
+
+
+#include "../OSCompatibilityLayer.h"
+
+
+
+namespace
+{
+
+bool PathIsReplaced(std::string_view path, const std::set<std::string>& replaced_paths)
+{
+	return std::ranges::any_of(replaced_paths, [&path](const std::string& replaced_path) {
+		if (path.starts_with(replaced_path))
```

Comment:
What if `path` is a substring of `replaced_path`, but actually a different folder, e.g.
```
path = common/culture_groups
replacedpath = common/culture
```
?

Reply:
Oh, shoot.

Reply:
Fixed